client/my-sites/checkout/src/components/domain-promotional-pricing-restrictions.tsx

```diff
@@ -0,0 +1,78 @@
+import { DOMAIN_PROMOTIONAL_PRICING_POLICY } from '@automattic/urls';
+import { localize } from 'i18n-calypso';
+import { Component } from 'react';
+import { gaRecordEvent } from 'calypso/lib/analytics/ga';
+import {
+	getDomainMappings,
+	getDomainRegistrations,
+	getDomainTransfers,
+	hasDomainRegistration,
+	hasTransferProduct,
+} from 'calypso/lib/cart-values/cart-items';
+import CheckoutTermsItem from 'calypso/my-sites/checkout/src/components/checkout-terms-item';
+import type { ResponseCart } from '@automattic/shopping-cart';
+import type { LocalizeProps } from 'i18n-calypso';
+
+export interface DomainPromotionalPricingRestrictionsProps {
+	cart: ResponseCart;
+}
+
+class DomainPromotionalPricingRestrictions extends Component<
+	DomainPromotionalPricingRestrictionsProps & LocalizeProps
+> {
+	recordPromotionalPricingPolicyClick = () => {
+		gaRecordEvent( 'Upgrades', 'Clicked Registration Agreement Link' );
+	};
+
+	hasAnyDomainWithPromotionalPrice( cart: ResponseCart ): boolean {
+		const domainProducts = [
+			...getDomainRegistrations( cart ),
+			...getDomainMappings( cart ),
+			...getDomainTransfers( cart ),
+		];
+		return domainProducts.some(
+			( product ) => product.item_subtotal_integer !== product.item_original_subtotal_integer
+		);
+	}
+
+	renderPromotionalPricingRestrictionsLink = () => {
+		return (
+			<p>
+				{ this.props.translate(
+					'{{promotionalPricingPolicyLink}}Restrictions apply{{/promotionalPricingPolicyLink}}.',
+					{
+						components: {
+							promotionalPricingPolicyLink: (
+								<a
+									href={ DOMAIN_PROMOTIONAL_PRICING_POLICY }
+									target="_blank"
+									rel="noopener noreferrer"
+									onClick={ this.recordPromotionalPricingPolicyClick }
+								/>
+							),
+						},
+					}
+				) }
+			</p>
+		);
+	};
+
+	render() {
+		const { cart } = this.props;
+		if (
+			! hasDomainRegistration( cart ) &&
+			! hasTransferProduct( cart ) &&
+			! this.hasAnyDomainWithPromotionalPrice( cart )
+		) {
```

Comment:
I believe this condition needs to be changed, otherwise the link will be shown for every cart that has a domain registration or transfer 🤔 I don't think it's a big deal to show it for every domain purchase since the support doc mentions promotional pricing explicitly, but I guess it's better to only show this for promotional domains
```suggestion
		const cartHasDomain = hasDomainRegistration( cart ) || hasTransferProduct( cart );
		if ( ! cartHasDomain || ! this.hasAnyDomainWithPromotionalPrice( cart ) ) {
```